tycho-ethereum/src/entrypoint_tracer/allowance_slot_detector.rs

```diff
@@ -704,73 +728,95 @@ impl EVMAllowanceSlotDetector {
             }
         }
 
-        for (idx, data) in validation_data.into_iter().enumerate() {
+        for (idx, mut metadata) in slots_to_test.into_iter().enumerate() {
             let response_id = (idx + 1) as u64;
 
             match id_to_response.get(&response_id) {
                 Some(response) => {
                     if let Some(error) = response.get("error") {
                         results.insert(
-                            data.token,
+                            metadata.token,
                             Err(AllowanceSlotError::RequestError(format!(
-                                "Validation call failed: {error}",
+                                "Slot test call failed: {error}",
                             ))),
                         );
                         continue;
                     }
 
+                    let (storage_addr, slot) = &metadata
+                        .all_slots
+                        .first()
+                        .expect("all_slots should not be empty")
+                        .0
+                        .clone();
+
                     match self.extract_allowance_from_call_response(response) {
                         Ok(returned_allowance) => {
-                            if returned_allowance != data.original_allowance {
+                            // Check if the override worked (allowance should be different from
+                            // original_allowance). We can't guarantee that it will match the
+                            // override value, as some tokens use shares
```

Comment:
I'm not sure this statement is valid for allowances, I think only for transfers slots.